### PR TITLE
Issue #9947 - check that each selector inside of  _selectors isn't null before calculating #totalKeys

### LINKED ISSUE #9947: Cannot invoke "org.eclipse.jetty.io.ManagedSelector.getTotalKeys()" because "selector" is null
**Jetty version**
Jetty 12 beta2

**Java version/vendor** `(use: java -version)`
openjdk version "17.0.7" 2023-04-18
OpenJDK Runtime Environment Temurin-17.0.7+7 (build 17.0.7+7)
OpenJDK 64-Bit Server VM Temurin-17.0.7+7 (build 17.0.7+7, mixed mode, sharing)

**OS type/version**
Windows 11 - Microsoft Windows [Version 10.0.22631.1900]

**Description**
_Do not report security issues here! See [Jetty Security Reports](https://www.eclipse.org/jetty/security_reports.php)._

**How to reproduce?**
Happens everytime we launch our server that embeds Jetty. Previous builds of Jetty 12 did not have this problem, or at least we didn't encounter it.

*Also: could someone give me a hint of where to find binary builds of Jetty 12 beta(x), e.g. nightly builds (also perhaps with some kind of indication as to how tests passed or not).*

The stack trace:
```
java.lang.NullPointerException: Cannot invoke "org.eclipse.jetty.io.ManagedSelector.getTotalKeys()" because "selector" is null
          	at org.eclipse.jetty.io.SelectorManager.getTotalKeys(SelectorManager.java:153)
          	at org.eclipse.jetty.io.SelectorManager.toString(SelectorManager.java:530)
          	at java.base/java.util.Formatter$FormatSpecifier.printString(Formatter.java:3147)
          	at java.base/java.util.Formatter$FormatSpecifier.print(Formatter.java:3025)
          	at java.base/java.util.Formatter.format(Formatter.java:2781)
          	at java.base/java.util.Formatter.format(Formatter.java:2717)
          	at java.base/java.lang.String.format(String.java:4155)
          	at org.eclipse.jetty.util.component.ContainerLifeCycle$Bean.toString(ContainerLifeCycle.java:791)
          	at com.iizix.Utilities.toString(Utilities.java:2569)
          	at com.iizix.Loggers.format(Loggers.java:606)
          	at com.iizix.Loggers.logInternalForced(Loggers.java:529)
          	at com.iizix.Loggers.logForced(Loggers.java:358)
          	at com.iizix.log.slf4j.LogLoggerImpl.debug(LogLoggerImpl.java:337)
          	at org.eclipse.jetty.util.component.ContainerLifeCycle.addBean(ContainerLifeCycle.java:419)
          	at org.eclipse.jetty.util.component.ContainerLifeCycle.addBean(ContainerLifeCycle.java:317)
          	at org.eclipse.jetty.quic.server.QuicServerConnector.<init>(QuicServerConnector.java:83)
          	at org.eclipse.jetty.http3.server.HTTP3ServerConnector.<init>(HTTP3ServerConnector.java:49)
          	at org.eclipse.jetty.http3.server.HTTP3ServerConnector.<init>(HTTP3ServerConnector.java:44)
```

For some reason, the `selector` variable is `null` during this initialization of the HTTP/3 connector.

### PR DESCRIPTION
**Pull request by strogiyotec**

### CODE REVIEW

File: jetty-io/src/main/java/org/eclipse/jetty/io/SelectorManager.java [hunk 1]
```diff
@@ -147,6 +147,10 @@ protected void execute(Runnable task)
     @ManagedAttribute(value = "Total number of keys in all selectors", readonly = true)
     public int getTotalKeys()
     {
+        if (_selectors == null || _selectors.length == 0)
```
Comment (sbordet) [inline jetty-io/src/main/java/org/eclipse/jetty/io/SelectorManager.java]: No, it's not `_selectors` that is `null`, it is the `selector` variable inside the loop below.

File: jetty-io/src/main/java/org/eclipse/jetty/io/SelectorManager.java [hunk 2]
```diff
@@ -147,6 +147,10 @@ protected void execute(Runnable task)
...
+        {
+            return 0;
+        }
         int keys = 0;
         for (final ManagedSelector selector : _selectors)
```
Comment (sbordet) [inline jetty-io/src/main/java/org/eclipse/jetty/io/SelectorManager.java]: Remove `final`, it is unnecessary boilerplate.

**Review by sbordet — CHANGES_REQUESTED** [review]

**Comment by strogiyotec** [comment]
@sbordet  thanks for the clarification, I addressed your comments

File: jetty-io/src/main/java/org/eclipse/jetty/io/SelectorManager.java [hunk 3]
```diff
@@ -147,14 +147,13 @@ protected void execute(Runnable task)
...
-        {
-            return 0;
-        }
         int keys = 0;
-        for (final ManagedSelector selector : _selectors)
+        for (ManagedSelector selector : _selectors)
         {
-            keys += selector.getTotalKeys();
+            if (selector != null)
+            {
```
Comment (sbordet) [inline jetty-io/src/main/java/org/eclipse/jetty/io/SelectorManager.java]: Please drop the braces for single line statements.

Also, there is an extra new line at the end of this method, please remove that too.

**Review by sbordet — CHANGES_REQUESTED** [review]

File: jetty-io/src/main/java/org/eclipse/jetty/io/SelectorManager.java (hunk 3 above)
Comment (strogiyotec) [reply jetty-io/src/main/java/org/eclipse/jetty/io/SelectorManager.java]: Done

**Review by sbordet — APPROVED** [review]

**Pull request merged**